src/pages/en/core-concepts/routing.md

```diff
@@ -142,6 +142,53 @@ In this example, a request for `/withastro/astro/tree/main/docs/public/favicon.s
 }
 ```
 
+#### Example: How to create an index.html at root level?
+
+If you want to create an index.html dynamically at rool level (e.g. for content fetched from a headless CMS), you have to provide `undefined` as value for the slug to `#getStaticPaths()`.
+
+```astro
+---
+// /src/pages/[...slug].astro
+export async function getStaticPaths() {
+  const pages = [
+    {
+      slug: undefined,
+      title: "Astro Store",
+      text: "Welcome to the Astro store!",
+    },
```

Comment:
```suggestion
#### Example: Dynamically create a top level root page

To create an index.html at root level (e.g. for content fetched from a headless CMS), include an object with `slug: undefined` in your `#getStaticPaths()` function.

```astro title="src/pages/[...slug].astro" "slug: undefined"
---
export async function getStaticPaths() {
  const pages = [
    {
      slug: undefined,
      title: "Astro Store",
      text: "Welcome to the Astro store!",
    },
```
This looks great @jablonski, thanks so much! I've made some minor adjustments to just tighten up the wording a bit and to add some code highlighting. Can you take a look and verify that it still expresses what you were going for?

Reply:
Perfect, thank you very much for improving my wording.